deploy-service/common/src/main/java/com/pinterest/deployservice/common/HTTPClient.java

```diff
@@ -41,33 +41,49 @@ private String generateUrlAndQuery(String url, Map<String, String> params) throw
         for (Map.Entry<String, String> entry : params.entrySet()) {
             sb.append(prefix);
             prefix = "&";
+            // note:  scrubUrlQueryValue could be expensive with many filtered values
+            //        consider using it only in only a DEBUG logging context in the future
+            String reportedValue = scrubUrl ? scrubUrlQueryValue(entry.getKey(), entry.getValue()) : entry.getValue();
```

Comment:
since changing existing method, please verify
1. call with scrubbed queries
2. call without scrubbed queries

Reply:
I think the unit tests below cover this already, did you mean something else?

Reply:
sure, UT is good verification. however, I was referring to Integ testing.